### DISCUSSION TITLE
GORILLA

### DISCUSSION THREAD

**Discussion by krystophny** [redmod-team/profit#119] [Use Cases]
From [GitHub](https://github.com/itpplasma/GORILLA):
> GORILLA computes guiding-center orbits for charged particles of given mass, charge and energy in toroidal fusion devices with three-dimensional field geometry. This guiding-center orbit code is designed to be used in scientific plasma physics simulations in the field of magnetic confinement fusion.

For us the computation of the *radial diffusion coefficient* `D11` is interesting to be explored with proFit.

### Inputs: 3
* Collisionality `nu_star`
* Energy `E`
* Electric field `v_E`
* *Radius?*

### Configuration
* number of particles `n_particles`
* particle species `species` (electron or deuterium)
* order of polynomial
* 3x grid density?
* Perturbation amplitude?

### Outputs: 1 + 1
* mono energetic radial diffusion (transport) coefficient `D11`
* standard deviation `D11_std`
* computation time?

## Open questions:
- [x] Include or avoid Matlab?
  - the executable can be launched directly with `profit`, Matlab is not needed
- [x] handle symbolic links in template and run directories
- [ ] How to deal with common preprocessing (flux tube volume, build)?
  - for now it was enough to do these steps manually
- [x] Categorical input data (run modes)
  - variable kind `Constant`
- [x] Interface: is ASCII file. (`numpytxt` postprocessor)
- [x] Avoid excessive date output: switch off diagnostics output.
- [ ] Reliablility checks (flux tube volume, number of lost particles).

**Comment by Rykath** [comment]
I've tested the internal version of GORILLA (which provides `mono_energetic_transp_main.x`) and set it up to work with proFit (pre `v0.4`) on *acluster* (TU Graz) and *VSC4* using [this](https://gist.github.com/Rykath/bd09eaa9b4c15eecbedb850e2129d3f7) configuration.
